src/Symfony/Component/Security/Http/Authenticator/AccessToken/HeaderAccessTokenExtractor.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\AccessToken;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Extracts a token from the request header.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ *
+ * @final
+ */
+class HeaderAccessTokenExtractor implements AccessTokenExtractorInterface
+{
+    public const HEADER_PARAMETER = 'Authorization';
+    public const BEARER_TOKEN_TYPE = 'Bearer';
+    private string $headerParameter;
+    private string $tokenType;
+
+    public function __construct(string $headerParameter = self::HEADER_PARAMETER, string $tokenType = self::BEARER_TOKEN_TYPE)
+    {
+        $this->headerParameter = $headerParameter;
+        $this->tokenType = $tokenType;
+    }
+
+    public function extractToken(Request $request): ?string
+    {
+        if (!$request->headers->has($this->headerParameter)) {
+            return null;
+        }
+        $header = $request->headers->get($this->headerParameter);
+        if (!\is_string($header)) {
+            return null;
+        }
+
+        if (1 === preg_match('/'.preg_quote($this->tokenType, '/').'\s*([a-zA-Z0-9\-_\+~\/\.]+)/', $header, $matches)) {
```

Comment:
shouldn't this be `\s+` rather than `\s*` ?

And shouldn't the regex be anchored ?

Reply:
but then, it means you need to add a space after non-empty token types. Because the space is mandatory in that case.

Reply:
Correct. The current code will allow `BearerACCESS_TOKEN` which should not be accepted.
I will adapt the regex.

Reply:
OK done with a test